lib/spack/spack/package_base.py

```diff
@@ -1090,15 +1090,6 @@ def update_external_dependencies(self, extendee_spec=None):
         """
         pass
 
-    def detect_dev_src_change(self):
-        """
-        Method for checking for source code changes to trigger rebuild/reinstall
-        """
-        dev_path_var = self.spec.variants.get("dev_path", None)
-        _, record = spack.store.STORE.db.query_by_spec_hash(self.spec.dag_hash())
-        mtime = fsys.last_modification_time_recursive(dev_path_var.value)
-        return mtime > record.installation_time
-
```

Comment:
I really don't want this deleted. I think it is wise to leave a hook at the package level so users can customize the behavior for their specific needs. I want this for my projects and I want to be able to tell users they can optimize change detection for their packages as they need.

This is also a breaking change for SNL's sierra code base.  We can port what we have for `git_sparse_path` packages, but these git detection models will likely break any package using that feature. For now it is just Sierra, but I envision this growing if/when we get more large monorepo projects willing to try out spack development

Reply:
To be clear, this PR would still allow packages to implement `detect_dev_src_change` (and if they do, it takes precedence over anything I've added here); this PR just changes how the default is encountered. Does that resolve your concern, or do you mean something else when you say

> This is also a breaking change for SNL's sierra code base.

?

Reply:
Ah yes. Sorry I must have missed the function relocation.